openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx

```diff
@@ -302,18 +328,28 @@ const Ingestion: React.FC<IngestionProps> = ({
     return element;
   };
 
-  const getSearchedIngestions = useCallback(() => {
+  const getSearchedIngestions = () => {
     const sText = lowerCase(searchText);
 
-    return sText
-      ? ingestionList.filter(
-          (ing) =>
-            lowerCase(ing.displayName).includes(sText) ||
-            lowerCase(ing.name).includes(sText)
-        )
-      : ingestionList;
+    setList(
+      sText
+        ? ingestionList.filter(
+            (ing) =>
+              lowerCase(ing.displayName).includes(sText) ||
+              lowerCase(ing.name).includes(sText)
+          )
+        : ingestionList
+    );
+  };
+
+  useEffect(() => {
+    getSearchedIngestions();
   }, [searchText, ingestionList]);
 
+  useEffect(() => {
+    fetchServicePermission();
+  }, [list]);
+
```

Comment:
Why are we chnaging the `getSearchedIngestions` function and storing the value in setList?
Every time on search if the list value changes then again we are fetching the permission why?

Reply:
@Sachin-chaurasiya  We are invoking the `getSearchedIngestions` function multiple times in the code, which is not required, as we set the value in state and use it wherever we want.
For the `Permission` one updating the code.

Reply:
Okay, make sure we are testing the search function properly.